### ISSUE TITLE
Enable old discovery method - what is the difference?

### ISSUE THREAD

**Issue by guschc** [home-assistant/home-assistant.github.io#35840]
### Feedback

Site tells you to enable the old discovery method in some cases. But I can't find any documentation on what the differences are between the old and new discovery methods. What disadvantages do I get by enabling the old method? There must be something, or it wouldn't have been replaced by a new method. Also, does changing the discovery method from new to old re-discover all devices, maybe ignoring name changes that I did and breaking automations?

### URL

https://www.home-assistant.io/integrations/fritz/

### Version

2024.11.2

**Comment by guschc** [comment]
I got to this because I am experiencing [this problem](https://github.com/home-assistant/core/issues/66887) with FritzOS 8.0

**Comment by mib1185** [comment]
The new (_and default used_) method is based on the mesh topology data of the fritzbox and has some advantages like you know via which mesh-device (_fritzobx or repeater_) your device is connected.
But this mesh topology data don't work correct on non-meshed setups or when there are 3rd party network devices involved (_eq. switches or non-AVM access points_).

**Comment by guschc** [comment]
Understood. So I have the choice of benefiting from [this fix](https://github.com/home-assistant/core/issues/128112) and being able to track the access point of my device, OR being able to have automations based on whether LAN devices are turned on or not.

Is it known if the fact that I have switches on my network will ruin the whole enter/leave detection of LAN devices or will it work as long as the LAN device is directly connected to a Fritzbox (or a secondary Fritzbox that is connected to the master via a switch)?

**Comment by mib1185** [comment]
Yes, the mentioned fix will only work for the new (_mesh topology based_) method.

Since I don't have such setup (_with multiple fritzboxes and switches_) available, I can't say for sure, how it will work - you've to test it. But I would guess, that direct to the Fritzbox connected devices, will be recognised correct.

Anyway, we use GitHub for tracking issues with Home Assistant itself, not for providing support - for any further questions, please try our [Community Forums](https://community.home-assistant.io) or join our [Discord chat server](https://www.home-assistant.io/join-chat).

Thanks! 👍

**Comment by guschc** [comment]
> Since I don't have such setup (_with multiple fritzboxes and switches_) available, I can't say for sure, how it will work - you've to test it. But I would guess, that direct to the Fritzbox connected devices, will be recognised correct.

I tried it out and am posting the results here, as they might be interesting for you, but also for anyone who gets here through Google when faced with a similar problem:

- New discovery method: Detection of a LAN device does **not** work, even if it is connected to a meshed secondary FritzBox that is connected to the master FritzBox via a switch
- Old discvoery method: Detection of such a device works, but the "away"-detection is delayed by one hour. The other way (device got connected) it is not delayed.
- Solution: The [PING](https://www.home-assistant.io/integrations/ping/) integration detects the state change in both directions reliably and fast 
- The old method caused for me some other problems: The master Fritzbox becomes unavailable from time to time, leading to flooding the Logbook with every single device that was ever discovered on the network becoming first unavailable and then within seconds changing to home or away again. This leads to several hundred entries. I saw that on average about once a day, it seems to be triggered by some changes on the network (e.g. unplugging and replugging a Fritz repeater, even though it is not between HA and the Master Fritzbox (HA runs on NAS directly connected to the Master Fritzbox)). I have noticed some inexplicable changes to "unavailable" on the new method as well, but it didn't cause log entries for every device. I will now switch back to the new method and report this there with more details once it happens again.

![Image](https://github.com/user-attachments/assets/2f56f018-3983-4c81-ac36-c14eee81f43e)

![Image](https://github.com/user-attachments/assets/a8ed96af-b64f-4f88-a63d-1756958d46b0)

![Image](https://github.com/user-attachments/assets/e63c4878-1f69-4f83-8458-0db35052ef61)

Bottom line: I cannot recommend switching to the old method for detection of LAN devices in a larger home network. Use Ping instead.
